### PR TITLE
Adding linter for formatting

### LINKED ISSUE #22: [FEATURE] Add linter
**Is your feature request related to a problem?**
Files have variable tab/spaces/empty lines/etc.

**What solution would you like?**
Add a linter that will auto-format models.

### PR DESCRIPTION
**Pull request by VachaShah**
### Description
* Adding linter for formatting
* Extracting smithy version to a separate file

### Issues Resolved
Closes #22 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
+	- [Adding a new API definition](#adding-a-new-api-definition)
+		- [Naming Convention](#naming-convention)
+		- [File Structure](#file-structure)
+		- [Defining the API model](#defining-the-api-model)
+	- [Adding a test-case for API definition](#adding-a-test-case-for-api-definition)
+		- [File Structure for Test-folder](#file-structure-for-test-folder)
+		- [Defining test-case for API model](#defining-test-case-for-api-model)
+	- [Local testing](#local-testing)
+		- [Pre-requisite](#pre-requisite)
+		- [Testing model API](#testing-model-api)
```
Comment (dblock) [inline DEVELOPER_GUIDE.md:17]: Are those tabs? :)

**Review by dblock — CHANGES_REQUESTED** [review]
We have 2 spaces everywhere in the project vs. tabs. You can see why, GitHub diff alone renders with a lot of spacing with tabs and is hard to read.

File: DEVELOPER_GUIDE.md (hunk 1 above)
Comment (VachaShah) [reply DEVELOPER_GUIDE.md:17]: Yeah :) I indented all the files in this repo with tabs including Smithy models. Should we go for spaces instead of tabs?

Comment (VachaShah) [reply DEVELOPER_GUIDE.md]: Changed it to indenting to spaces.

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -119,14 +128,14 @@ Let's suppose we have test-cases for put mapping and search api at first.
...
-  ```
-  test
-  ├── scripts
-  └── model
+```
+
+We'd want to include the *Index-Aliases API* now.
+The project-tree structure will be as follows:
+```
+test
```
Comment (pgtgrly) [inline DEVELOPER_GUIDE.md:136]: [nit] the formatting of the directory tree seems to be a bit off, can you correct that?

File: model/_global/ping/operations.smithy [hunk 3]
```diff
@@ -11,8 +11,7 @@ namespace OpenSearch
 @suppress(["HttpUriConflict"])
 @http(method: "GET", uri: "/")
 @documentation("Returns whether the cluster is running.")
-operation GetPingCluster {
-    input: GetPingClusterInput,
-    output: GetPingClusterOutput
+operation GetClusterInfo {
```
Comment (sachetalva) [inline model/_global/ping/operations.smithy:14]: can we rename the directory also from `ping` to `get_cluster_info`?

File: DEVELOPER_GUIDE.md (hunk 2 above)
Comment (VachaShah) [reply DEVELOPER_GUIDE.md:136]: Corrected.

File: model/_global/ping/operations.smithy (hunk 3 above)
Comment (VachaShah) [reply model/_global/ping/operations.smithy:14]: Done.

**Review by pgtgrly — APPROVED** [review]

**Review by sachetalva — APPROVED** [review]

**Pull request merged**